docs/cep/table/table-indexes.md

```diff
@@ -3,34 +3,172 @@ sidebar_position: 30
 title: Table Indexes
 ---
 
-Indexes allow tables to be searched/modified much faster. For more information about indexing tables (collections), refer to [Indexing](../../collections/indexing/index.md).
+Macrometa stream workers provide various indexing options to help you efficiently search and modify data in tables (collections). This page explains how to create indexes using `CREATE INDEX` in Macrometa stream workers.
 
-Indexes can be configured together with primary keys. When more than one attribute is used for index, each one of them is used to index the table for fast access to the data.
+## Purpose
 
-## Example 1
+Indexes allow tables to be searched and modified much faster. Indexes can be configured together with primary keys. When more than one attribute is used for index, each one of them is used to index the table for fast access to the data.
+
+For more information about indexing tables (collections), refer to [Indexing](../../collections/indexing/).
+
+## Syntax
+
+There are two ways to create an index.
+
+### CREATE INDEX
+
+You can use the `CREATE INDEX` command on a Macrometa table (collection) that your stream worker uses as a source or store:
 
 ```sql
-CREATE SOURCE StockTable WITH (type='database',collection='StockTable',PrimaryKey='symbol', Index='key1', Index='key2') (symbol string, price float, volume long);
+CREATE (UNIQUE)? INDEX index-name ON TABLE table-name [WITH (type="<type>", optional properties...)] (fields_to_index...)
 ```
 
-## Example 2
+The following parameters are configured in an index definition:
+
+| Parameter       | Description                                               |
+|-----------------|-----------------------------------------------------------|
+| (UNIQUE)?       | Optional flag to create unique indexes.                   |
+| index-name      | Name for the index.                                       |
+| table-name      | Name of the table to create indexes.                      |
+| fields_to_index | Comma separated list of 1:* index fields without types (i.e id, name). |
+
+### CREATE SOURCE with an Index
+
+You can incorporate it into a source definition for a `database` (Macrometa collection) source. For more information about source definitions, refer to [Sources](../source/index.md).
 
 ```sql
--- Creates a persistent index named `SamplePersistentIndex` on `SampleGDNTable` with following properties {unique=true, sparse=true, deduplicate=true}.
-CREATE UNIQUE INDEX SamplePersistentIndex ON TABLE SampleGDNTable WITH(type="persistent", sparse="true", deduplicate="true") (sensorId);
+CREATE SOURCE <source_name> WITH (type="database", collection="<collection>", PrimaryKey="<primary-key-field>", Index="<index-name-1>", Index="<index-name-2>"...) (<attribute_name> <attribute_type>, <attribute_name> <attribute_type>...);
```

Comment:
@StoyanTc, what kind of index is created if this method is used? Can you specify an index type with this method?

Reply:
@diana-macrometa - This is related for a RDBMS table. The `c8db` collection does not have explicit PK. I would suggest to remove the section `CREATE SOURCE with an Index`.